### PR TITLE
Add a way to quickly check NPC's stats when assign workers to a camp or update workers jobs

### PR DESCRIPTION
**Pull request by Godofbush**
#### Summary

SUMMARY: Interface "add a way to quickly check NPC's stats when assign workers to a camp or update workers jobs."

#### Purpose of change

Make life easier. Happy managing faction camp. #46687 

#### Describe the solution

Press `i` to inspect current selected NPC.

#### Describe alternatives you've considered

None

#### Testing

1. Create a new world.
2. Spawn NPC as many as you can, in my case, 21 NPCs.
3. Ask one of NPCs to start a faction camp.
4. Examining the bulletin borad and select `Assign Workers` mission.
5. Check the interface: should has a tip on the bottom, and scrolling should behave normally.
6. Assign all workers to the camp, and check the interface on jobs panel.

#### Additional context

image

image

### CODE REVIEW

File: src/faction_camp.cpp [hunk 1]
```diff
@@ -1698,7 +1702,10 @@ void basecamp::worker_assignment_ui()
 
         ui_manager::redraw();
         const std::string action = ctxt.handle_input();
-        if( action == "DOWN" ) {
+        std::string sInput = ctxt.get_raw_input().text;
+        if (sInput == "i") {
```
Comment (Qrox) [inline src/faction_camp.cpp:1706]: This should be added as an action so player can change the keybindings. (Edit `data/raw/keybindings.json`)

File: src/faction_camp.cpp [hunk 2]
```diff
@@ -1675,6 +1677,8 @@ void basecamp::worker_assignment_ui()
         } else {
             mvwprintz( w_followers, point( 1, 4 ), c_light_red, no_npcs );
         }
+        mvwprintz( w_followers, point( 1, FULL_SCREEN_HEIGHT - 2 ), c_light_gray,
+                   _("Press i to inspect this follower."));
```
Comment (Qrox) [inline src/faction_camp.cpp:1681]: Use `ctxt.get_desc( "<ACTION>" )` to get the keybinding description on the fly.

File: src/faction_camp.cpp (hunk 1 above)
Comment (BreadPrince) [reply src/faction_camp.cpp]: Done, thx

File: src/faction_camp.cpp (hunk 2 above)
Comment (BreadPrince) [reply src/faction_camp.cpp]: Done, thx

File: data/raw/keybindings.json [hunk 3]
```diff
@@ -3428,5 +3428,12 @@
     "id": "CHECK_OPINION",
     "name": "Check opinion",
     "bindings": [ { "input_method": "keyboard_char", "key": "O" }, { "input_method": "keyboard_code", "key": "o", "mod": [ "shift" ] } ]
+  },
+  {
+    "type": "keybinding",
+    "category": "FACTION_MANAGER",
+    "id": "INSPECT_NPC",
+    "name": "Inspect NPC",
+    "bindings": [ { "input_method": "keyboard_char", "key": "i" } ]
```
Comment (Qrox) [inline data/raw/keybindings.json:3437]:
```suggestion
    "bindings": [ { "input_method": "keyboard_any", "key": "i" } ]
```
Add keybindings for both symbol and keycode input modes.

**Comment by Qrox** [comment]
Also please astyle your changes.

**Comment by BreadPrince** [comment]
> Also please astyle your changes.

I'll try, I'm new to C++, no so familar with the toolchain.

Thanks for your reviewing.

File: src/faction_camp.cpp [hunk 4]
```diff
@@ -1698,7 +1701,9 @@ void basecamp::worker_assignment_ui()
 
         ui_manager::redraw();
         const std::string action = ctxt.handle_input();
-        if( action == "DOWN" ) {
+        if( action == "INSPECT_NPC" ) {
+            cur_npc->disp_info();
```
Comment (BrettDong) [inline src/faction_camp.cpp:1705]:
```suggestion
            if( cur_npc ) {
                cur_npc->disp_info();
            }
```

File: src/faction_camp.cpp [hunk 5]
```diff
@@ -1810,7 +1818,9 @@ void basecamp::job_assignment_ui()
         ui_manager::redraw();
 
         const std::string action = ctxt.handle_input();
-        if( action == "DOWN" ) {
+        if( action == "INSPECT_NPC" ) {
+            cur_npc->disp_info();
```
Comment (BrettDong) [inline src/faction_camp.cpp:1822]:
```suggestion
            if( cur_npc ) {
                cur_npc->disp_info();
            }
```

**Review by BrettDong — COMMENTED** [review]
Try to appease `clang-tidy` warnings.

**Review by BrettDong — CHANGES_REQUESTED** [review]
1. Spawn a NPC and start a camp in a new world
2. Enter "Assign Workers" UI
3. Press **ENTER** to assign a NPC
4. Now it says "You have no companions following you."
5. Press **i**
6. Game crashes

**Review by BrettDong — APPROVED** [review]

**Pull request merged**